kratos/utilities/entities_utilities.cpp

```diff
@@ -12,16 +12,199 @@
+    const auto& r_entity = KratosComponents<TEntity>::Get(rName);
+    mpPrototypeEntity = r_entity.Create(0, r_entity.pGetGeometry(), nullptr);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+void EntitityIdentifier<TEntity>::GenerateMultipleTypes(const std::string& rName)
+{
+    // Check the type of entity considered
+    const std::string entity_type_name = GetEntityTypeName();
+
+    // Split the name
+    const auto splitted_names = StringUtilities::SplitStringByDelimiter(rName, ';');
+    for (auto& r_entity_name : splitted_names) {
+        // Some values need to be mandatorily prescribed since no meaningful default value exist. For this reason try accessing to them so that an error is thrown if they don't exist
+        KRATOS_ERROR_IF(r_entity_name != "" && !KratosComponents<TEntity>::Has(r_entity_name)) << entity_type_name << " name not found in KratosComponents<" << entity_type_name << "> -- name is " << r_entity_name << std::endl;
+        const auto& r_ref_entity = KratosComponents<TEntity>::Get(r_entity_name);
+        const auto& r_reference_geometry = r_ref_entity.GetGeometry();
+        const auto& r_reference_geometry_type = r_reference_geometry.GetGeometryType();
+        mTypes.insert({r_reference_geometry_type, &KratosComponents<TEntity>::Get(r_entity_name)});
+    }
+
+    // Set the definition type
+    mDefinitionType = EntitiesUtilities::DefinitionType::Multiple;
+
+    // Assign default prototype
+    const auto* p_entity = mTypes.begin()->second;
+    mpPrototypeEntity = p_entity->Create(0, p_entity->pGetGeometry(), nullptr);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+void EntitityIdentifier<TEntity>::GenerateTemplatedTypes(const std::string& rName)
+{
+    // Prepare regex 
+    const std::string replace_dimension = StringUtilities::ReplaceAllSubstrings(rName, "#D", "(2D|3D)");
+    const std::string replace_number_of_nodes = StringUtilities::ReplaceAllSubstrings(replace_dimension, "#N", "[0-9]+N");
+    std::regex pattern(replace_number_of_nodes);
+
+    // Geometry combinations
+    for (const auto& r_pair : KratosComponents<TEntity>::GetComponents()) {
+        const std::string& r_key = r_pair.first;
+        if (std::regex_match(r_key, pattern)) {
+            const auto& r_entity = KratosComponents<TEntity>::Get(r_key);
+            mTypes.insert({r_entity.GetGeometry().GetGeometryType(), &r_entity});
+        }
+    }
+
+    // Set the definition type
+    mDefinitionType = EntitiesUtilities::DefinitionType::Multiple;
+
+    // Assign default prototype
+    const auto* p_entity = mTypes.begin()->second;
+    mpPrototypeEntity = p_entity->Create(0, p_entity->pGetGeometry(), nullptr);
```

Comment:
here it must be a nullptr...

Reply:
No, as I said if nullptr it will fail later when checking the geometry type is the same as the buffer (which is required to minimize the number of searches in the map)